pkgs/data/fonts/nerdfonts-patcher/default.nix

```diff
@@ -0,0 +1,52 @@
+{ stdenv
+, fetchFromGitHub
+, lib
+, python3
+, fontforge
+
+, patchFont ? null
+, fontName ? null
+, glyphFlags ? "-c"
+}:
+
+assert (patchFont != null);
+assert (fontName != null);
+
+stdenv.mkDerivation rec {
+  pname = fontName + "-nerd-font";
+  version = "v2.1.0";
+  
+  src = fetchFromGitHub {
+    owner = "ryanoasis";
+    repo = "nerd-fonts";
+    rev = version;
+    sha256 = "sha256-j0cLZHOLyoItdvLOQVQVurY3ARtndv0FZTynaYJPR9E=";
+  };
+
+  nativeBuildInputs = [
+    python3
+    fontforge
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+    mkdir -p $out/share/fonts
+    find ${toString patchFont}/share/fonts/**/*.ttf -exec python font-patcher ${glyphFlags} -o $out/share/fonts/ {} \;
```

Comment:
The modern format otf is also supported.

Reply:
Good one. I'll fix this tomorrow